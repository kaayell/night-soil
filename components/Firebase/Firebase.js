import * as firebase from 'firebase'
import { ANDROID_API_KEY, config, IOS_API_KEY } from './secrets'

export default class Firebase {
  static init () {
    firebase.initializeApp(config)
  }

  static getAuth() {
    return firebase.auth()
  }

  static getUserDetailsRef() {
    let uid = firebase.auth().currentUser.uid
    return firebase.database().ref(`/user/details/${uid}`)
  }

  static savePoop(poopDetails) {
    let currentUser = firebase.auth().currentUser
    firebase.database().ref(`/poop/${currentUser.uid}`).push().set(poopDetails)
  }

  static getPoopsRef(){
    let uid = firebase.auth().currentUser.uid
    return firebase.database().ref(`/poop/${uid}`)
  }

  static async loginWithGoogle () {
    const response = await Expo.Google.logInAsync({
      androidClientId: ANDROID_API_KEY,
      iosClientId: IOS_API_KEY,
      scopes: ['profile', 'email'],
    })

    if (response.type === 'success') {
      const credential = firebase.auth.GoogleAuthProvider.credential(response.idToken, response.accessToken)
      firebase.auth().signInWithCredential(credential)
    }
  }
}