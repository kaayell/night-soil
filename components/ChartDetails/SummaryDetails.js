import React, {Component} from 'react'
import {Platform, ScrollView, StatusBar, Text, TouchableOpacity, View} from "react-native";
import Firebase from "../Firebase/Firebase";
import {Icon, List, ListItem} from "react-native-elements";
import {BLUE, OFF_WHITE} from "../StyleGuide/colors";
import _ from "lodash";
import {POPPINS} from "../StyleGuide/fonts";
import StarRating from "react-native-star-rating";

class SummaryDetails extends Component {
  constructor(props) {
    super(props)

    this.state = {
      poopData: []
    }
  }

  componentDidMount() {
    Firebase.getPoopsRef().on('value', snapshot => {
      let poopData = []
      snapshot.forEach(childSnapshot => {
        const poop = childSnapshot.val()
        poopData.push(poop)
      })
      this.setState({poopData})
    })
  }

  render() {
    const howMuchMade = _.sum(_.flatMap(this.state.poopData, (poop) => {
      return poop.atWork && ((poop.salary / 60) * parseInt(poop.durationMinutes))
    }))

    const averageTime = _.mean(_.flatMap(this.state.poopData, (poop) => {
      return parseInt(poop.durationMinutes)
    }))

    const paddingTop = Platform.OS === 'ios' ? (parseInt(Platform.Version, 10) > 10 ? 35 : 20) : StatusBar.currentHeight + 10
    return (
      <View style={{flex: 1, backgroundColor: OFF_WHITE, paddingTop, height: "100%"}}>
        <TouchableOpacity style={{flexDirection: 'row', height: '5%', justifyContent: 'flex-end', paddingRight: 15}}
                          onPress={() => this.props.navigation.goBack()}>
          <Icon name={'close'} color={BLUE}/>
        </TouchableOpacity>
        <View style={{
          marginTop: 20,
          justifyContent: 'center',
          width: "95%",
          height: 200,
          backgroundColor: 'white',
          shadowOpacity: 0.5,
          shadowRadius: 5,
          shadowColor: BLUE,
          shadowOffset: {height: 0, width: 0},
          alignSelf: 'center'
        }}>
          <View style={{flex: 1, flexDirection: 'row'}}>
            <View style={{flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
              <Text style={{fontFamily: POPPINS}}>MONEY MADE POOPIN</Text>
              <Text style={{fontFamily: POPPINS, fontSize: 40, color: 'green'}}>${howMuchMade.toFixed(2)}</Text>
            </View>
            <View style={{flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
              <Text style={{fontFamily: POPPINS}}>POOPS RECORDED</Text>
              <Text style={{fontFamily: POPPINS, fontSize: 40, color: 'green'}}>{this.state.poopData.length}</Text>
            </View>
          </View>
          <View style={{flex: 1, flexDirection: 'row'}}>
            <View style={{flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
              <Text style={{fontFamily: POPPINS}}>AVERAGE TIME SPENT</Text>
              <Text style={{fontFamily: POPPINS, fontSize: 40, color: 'green'}}>
                {averageTime.toFixed(0)}<Text style={{fontSize: 20}}>min</Text></Text>
            </View>
          </View>
        </View>
        <ScrollView>
          <List containerStyle={{marginBottom: 20}}>
            {
              this.state.poopData.slice(0).reverse().map((data, i) => (
                <ListItem
                  key={i}
                  title={
                    <StarRating
                      maxStars={5}
                      disabled={true}
                      starSize={20}
                      rating={data.poopRating}
                      fullStarColor={"#FFCC00"}
                      containerStyle={{alignItems: 'center', width: 80, paddingLeft: 10}}
                    />}
                  containerStyle={{padding: 10, height: 60}}
                  hideChevron={true}
                  disabled={true}
                  fontFamily={POPPINS}
                  subtitle={`${data.durationMinutes} min`}
                  rightTitle={data.date}
                  topDivider={false}
                />
              ))
            }
          </List>
        </ScrollView>
      </View>
    )
  }
}

export default SummaryDetails