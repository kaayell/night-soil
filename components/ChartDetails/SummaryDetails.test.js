import React from "react";
import {shallow} from "enzyme";
import SummaryDetails from "./SummaryDetails";
import Firebase from "../Firebase/Firebase";

describe('SummaryDetails', () => {
  it('should render', () => {
    Firebase.getPoopsRef = jest.fn(() => {
      return {
        on: jest.fn(() => Promise.resolve(
          [
            {
              val: () => {
                return "hi"
              }
            },
          ]
        ))
      }
    })
    const wrapper = shallow(<SummaryDetails />)
    expect(wrapper).toMatchSnapshot()
  });
});