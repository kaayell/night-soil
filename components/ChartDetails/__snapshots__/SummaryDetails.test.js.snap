// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryDetails should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "height": "100%",
      "paddingTop": 20,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "height": "5%",
        "justifyContent": "flex-end",
        "paddingRight": 15,
      }
    }
  >
    <Icon
      color="#9AC0CD"
      name="close"
      raised={false}
      reverse={false}
      reverseColor="white"
      size={24}
      underlayColor="white"
    />
  </TouchableOpacity>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "height": 200,
        "justifyContent": "center",
        "marginTop": 20,
        "shadowColor": "#9AC0CD",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
        "width": "95%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "poppins-regular",
            }
          }
        >
          MONEY MADE POOPIN
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "green",
              "fontFamily": "poppins-regular",
              "fontSize": 40,
            }
          }
        >
          $
          0.00
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "poppins-regular",
            }
          }
        >
          POOPS RECORDED
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "green",
              "fontFamily": "poppins-regular",
              "fontSize": 40,
            }
          }
        >
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "poppins-regular",
            }
          }
        >
          AVERAGE TIME SPENT
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "green",
              "fontFamily": "poppins-regular",
              "fontSize": 40,
            }
          }
        >
          0
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
              }
            }
          >
            min
          </Text>
        </Text>
      </View>
    </View>
  </View>
  <ScrollView>
    <List
      containerStyle={
        Object {
          "marginBottom": 20,
        }
      }
    />
  </ScrollView>
</View>
`;
